src/main/java/edu/cornell/kfs/fp/batch/CorporateBilledCorporatePaidLoadFlatFileStep.java

```diff
@@ -0,0 +1,77 @@
+package edu.cornell.kfs.fp.batch;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.fp.batch.service.ProcurementCardLoadTransactionsService;
+import org.kuali.kfs.sys.batch.AbstractStep;
+import org.kuali.kfs.sys.batch.BatchInputFileType;
+import org.kuali.kfs.sys.batch.service.BatchInputFileService;
+import org.kuali.kfs.sys.batch.service.WrappingBatchService;
+import org.kuali.kfs.sys.service.ReportWriterService;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+
+public class CorporateBilledCorporatePaidLoadFlatFileStep extends AbstractStep {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CorporateBilledCorporatePaidLoadFlatFileStep.class);
+    private ProcurementCardLoadTransactionsService procurementCardLoadTransactionsService;
+    private BatchInputFileService batchInputFileService;
+    private BatchInputFileType corporateBilledCorporatePaidInputFileType;
+    private ReportWriterService reportWriterService;
+    
+    public boolean execute(String jobName, Date jobRunDate) {
+        procurementCardLoadTransactionsService.cleanTransactionsTable();
+
+        List<String> fileNamesToLoad = batchInputFileService.listInputFileNamesWithDoneFile(corporateBilledCorporatePaidInputFileType);
+        
+        ((WrappingBatchService) reportWriterService).initialize();
+        
+        List<String> processedFiles = new ArrayList<String>();
+        for (String inputFileName : fileNamesToLoad) {
+            LOG.info("execute, file name: " + inputFileName);
+            try {
+                boolean currentFileProccessed = procurementCardLoadTransactionsService.loadProcurementCardFile(inputFileName, reportWriterService);
+                if (currentFileProccessed) {
+                    processedFiles.add(inputFileName);
+                }
+                
+            } catch (RuntimeException e) {
+                LOG.error("execute, There was an error proccessing " + inputFileName, e);
+            }
+        }
+
+        ((WrappingBatchService) reportWriterService).destroy();
+
+        removeDoneFiles(processedFiles);
+
+        return true;
+    }
+
+    private void removeDoneFiles(List<String> dataFileNames) {
+        for (String dataFileName : dataFileNames) {
```

Comment:
Is this specific method needed? Or could the FileStorageService be used for removing the .done files?

Reply:
let me check for a cleaner way using the pcard document job style, I based this code off of that.